### PR TITLE
Transcribe recording functionality

### PR DESCRIPTION
**Pull request by piraka9011**
Enable the transcribe server to record audio data.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -123,18 +123,22 @@ io.on('connection', (socket) => {
...
         }
-
         // Propagate raw transcript to client
         socket.emit('speechResult', {
             'text': transcript,
             'isFinal': true
         });
 
         socket.transcriber.onTranscript(transcript, true);
+        let result = sendRecording(
```
Comment (Ananasship) [inline src/index.js]: Can we do this asynchronously?

File: src/index.js [hunk 2]
```diff
@@ -123,18 +123,22 @@ io.on('connection', (socket) => {
...
+          audioBuffer,
+          socket.transcriber.currentSurahNum,
+          socket.transcriber.currentAyahNum);
+        console.log(`sendRecording result: ${result}`)
```
Comment (Ananasship) [inline src/index.js]: we can remove this console log since we're logging through the function call.

**Comment by fdalvi** [comment]
I will look at the commit in detail inshaAllah, but a high level comment I have is that we should keep the speech processing logic separate from the audio upload logic - this is what (imo the badly names) `fileHandler.js` module was originally supposed to do. If you see the `sendStream` event, we pipe the audio to both the speech handler, and the file handler (and the file handler would do all the audio buffering etc while the speech handler is oblivious of this upload handling code). What do you think @Ananasship?

**Comment by piraka9011** [comment]
> If you see the `sendStream` event, we pipe the audio to both the speech handler, and the file handler.

So I guess I should emit an `upload` event when we receive an `onFinal`?

**Comment by fdalvi** [comment]
> So I guess I should emit an `upload` event when we receive an `onFinal`?

Do you mean an `upload` event from the client? I think the client should also be oblivious of this. All you have to do is move the audioBuffer code from `speech.js` to `fileHandler.js` (and potentially rename this file to something that makes more sense). `index.js` will then have the code to 1: send audio to both speech/fileHandler, and 2: the onFinal handler in `index.js` will call some function in fileHandler to cause the actual upload.

Separate from the above issue, the other thing is that by the time you get a onFinal event, there are multiple possible scenarios:
1: we have already started reciting the next ayah (so audio buffer contains some part of next ayah)
2: we are in the middle of an ayah (onFinal can happen anywhere google decides, in this case audio buffer has a partial ayah)
3: we have recited > 1 ayat without an onFinal (sometimes an onFinal occurs once in 5-6 ayat if there are no discernible pauses, in this case audio buffer has multiple ayat)

So in my opinion, onFinal really is not a useful segmenting point that makes sense in our context, and so the decision to upload etc should not depend on onFinal. One suggestion is every "session" (connect ----> disconnect or startStream -----> endStream) have a single audio upload, and some metadata about this upload (maybe in the filename, maybe elsewhere) contain all the ayahFound results for this "session"

**Pull request closed without merging**